### ISSUE TITLE
Backup S3-compatible bucket using Rustic

### ISSUE THREAD

**Issue by Wouter0100** [rustic-rs/rustic#1278]
It would be really nice to, instead of a filesystem or stdin, backup a S3-compatible bucket using rustic. 

I've tried various techniques (rclone mount for example) and then read that mount by rustic, but it's very inefficient in deciding what files are changed and results in a lot of API calls to S3. I think it should be able to do it more efficient.

**Comment by bebehei** [comment]
hey @Wouter0100 I'm having the same issue, I'm currently testing it with Hetzner S3 storeage and rustic 0.7.

You might want to look at #1044 and #1045.

Apparently the `s3:...` endpoint is canceled and `opendal:s3` should be used. I'm currently testing this while writing this comment.

The docs right now are confusing, because there is no separate section in the docs, but ["Other services using rclone"](https://rustic.cli.rs/docs/commands/init/rclone.html) pointing into the direction of rclone, but not opendal.

**Comment by simonsan** [comment]
> The docs right now are confusing, because there is no separate section in the docs, but ["Other services using rclone"](https://rustic.cli.rs/docs/commands/init/rclone.html) pointing into the direction of rclone, but not opendal.

That's right, I think that is missing, and I started to work on it in: https://github.com/rustic-rs/docs/pull/90

**Comment by aawsome** [comment]
Note that we are talking about two things here: opendal providing access to the repository and opendal as a source to run backups on (which will then be stored in a repository, independent from how to access this).

For the first, most is implemented but docu and edge case handling needs improvement.
For the second, we don't have an implementation yet.

**Comment by Wouter0100** [comment]
Yes, my feature request was for the second part. Being able to backup a S3 bucket (for example, using opendal) towards a rustic repository _somewhere_.

**Comment by bebehei** [comment]
I just had been successful adding a S3 repo into my storage. Thanks for fixing this so fast!

Docs seem to be very fine. There is a copy-pastable example and link to the reference. Thats everything you need, when you start fresh.

**Comment by sanmai-NL** [comment]
It's unclear to me whether this feature is at thistime implemented completely or just partially.

**Comment by aawsome** [comment]
@sanmai-NL The feature to use S3 as a backup source is not yet implemented - this is why this issue is still open.
A workaround is to use a tool to mount your S3 and than backup that mounted directory.

**Comment by Wouter0100** [comment]
Unfortunately, for larger buckets that's not really usable generally, that's why I opened this feature request :)
